sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java

```diff
@@ -348,6 +348,8 @@ static Object singularFieldToProtoValue(
           return Double.valueOf((String) value);
         } else if (value instanceof Double || value instanceof Float) {
           return ((Number) value).doubleValue();
+        } else if (value instanceof Integer || value instanceof Long) {
+          return ((Number) value).longValue();
```

Comment:
Shouldn't this still be .doubleValue() since the proto schema for FLOAT will be a double? If so, then maybe we can simply replace

else if (value instanceof Double || value instanceof Float)

with

else if (value instanceof Number)

Reply:
I was not totally sure if int->double conversion will be accurate, but I guess we will have to do the conversion here or there. Done.